web/modules/custom/stats/joinup_stats.module

```diff
@@ -0,0 +1,138 @@
+<?php
+
+/**
+ * @file
+ * Joinup Statistics module main file.
+ */
+
+declare(strict_types = 1);
+
+use Drupal\Core\Entity\ContentEntityInterface;
+use Drupal\Core\Entity\EntityInterface;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\Core\Site\Settings;
+use Drupal\meta_entity\Entity\MetaEntity;
+
+/**
+ * Implements hook_entity_insert().
+ *
+ * Ensure meta entities for new content entities configured to have attached
+ * metadata.
+ */
+function joinup_stats_entity_insert(EntityInterface $entity): void {
+  if ($entity instanceof ContentEntityInterface) {
+    /** @var \Drupal\meta_entity\MetaEntityRepositoryInterface $repository */
+    $repository = \Drupal::service('meta_entity.repository');
+    foreach ($repository->getMetaEntityTypesForBundle($entity->getEntityTypeId(), $entity->bundle()) as $type) {
+      $meta_entity = MetaEntity::loadOrCreate($type, $entity);
+      if ($meta_entity->isNew()) {
+        $meta_entity->save();
```

Comment:
This is clear to me now. We need to have a meta entity available on creating because we are tracking the download counts in the meta entity using a cached computed field. This field has an expiration time after which it will be refreshed. If we do not have the field, then we can not know when it expires, and it will never be updated.